src/modules/hip/kernel/log.hpp

```diff
@@ -0,0 +1,114 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+template <typename T>
+__device__ void log_hip_compute(T *srcPtr, d_float8 *src_f8, d_float8 *dst_f8)
+{
+    if constexpr (std::is_same<T, schar>::value)
+        rpp_hip_math_add8_const(src_f8, src_f8, (float4)128);
+
+    dst_f8->f1[0] = __logf(fabsf(src_f8->f1[0]));
+    dst_f8->f1[1] = __logf(fabsf(src_f8->f1[1]));
+    dst_f8->f1[2] = __logf(fabsf(src_f8->f1[2]));
+    dst_f8->f1[3] = __logf(fabsf(src_f8->f1[3]));
+    dst_f8->f1[4] = __logf(fabsf(src_f8->f1[4]));
+    dst_f8->f1[5] = __logf(fabsf(src_f8->f1[5]));
+    dst_f8->f1[6] = __logf(fabsf(src_f8->f1[6]));
+    dst_f8->f1[7] = __logf(fabsf(src_f8->f1[7]));
+
+    if constexpr (std::is_same<T, schar>::value)
+    {
+        dst_f8->f4[0] = rpp_hip_pixel_check_0to255(dst_f8->f4[0]) - (float4)128;
+        dst_f8->f4[1] = rpp_hip_pixel_check_0to255(dst_f8->f4[1]) - (float4)128;
+    }
+}
+
+// F16 stores without layout toggle (8 F16 pixels)
+
+__device__ __forceinline__ void rpp_hip_pack_float8_and_store(half *dstPtr, d_float8 *dst_f8)
+{
+    d_half8 dst_h8;
+
+    dst_h8.h2[0] = __float22half2_rn(make_float2(dst_f8->f1[0], dst_f8->f1[1]));
+    dst_h8.h2[1] = __float22half2_rn(make_float2(dst_f8->f1[2], dst_f8->f1[3]));
+    dst_h8.h2[2] = __float22half2_rn(make_float2(dst_f8->f1[4], dst_f8->f1[5]));
+    dst_h8.h2[3] = __float22half2_rn(make_float2(dst_f8->f1[6], dst_f8->f1[7]));
+
+    *(d_half8 *)dstPtr = dst_h8;
+}
+
+// F32 stores without layout toggle (8 F32 pixels)
+
+template <typename T>
+__device__ __forceinline__ void rpp_hip_pack_float8_and_store(T *dstPtr, d_float8 *dst_f8)
+{
+    *(d_float8_s *)dstPtr = *(d_float8_s *)dst_f8;
```

Comment:
Why again this function is added here?
It is already available in rpp_hip_common.hpp

Reply:
This takes in template as input as in here it supports Rpp8u, Rpp8s and Rpp16f but this functionality is different in rpp_hip_common.hpp

Will modify function name to avoid confusion with existing function.

Reply:
Why template argument is needed for this?
I am still a little unclear on why an extra function is needed

Are all the outputs are float? even for F16 is it float?

Reply:
for f16 alone its float16 outputs but for all others its float32.